dpnp/backend/kernels/dpnp_krnl_elemwise.cpp

```diff
@@ -38,210 +38,277 @@
     template <typename _KernelNameSpecialization1, typename _KernelNameSpecialization2>                                \
     class __name__##_kernel;                                                                                           \
                                                                                                                        \
-    template <typename _DataType_input, typename _DataType_output>                                                     \
-    void __name__(void* array1_in, void* result1, size_t size)                                                         \
+    template <typename _KernelNameSpecialization1, typename _KernelNameSpecialization2>                                \
+    class __name__##_strides_kernel;                                                                                   \
+                                                                                                                       \
+    template <typename _DataType_output, typename _DataType_input>                                                     \
```

Comment:
Why we need this renaming/switching of order of typenames in template?
`template <typename _DataType_input, typename _DataType_output>` -> `template <typename _DataType_output, typename _DataType_input>`
It looks like significant part of changes in this PR are related to this renaming.